pinecone/client.go

```diff
@@ -6,17 +6,34 @@ import (
 	"fmt"
 	"github.com/deepmap/oapi-codegen/v2/pkg/securityprovider"
 	"github.com/pinecone-io/go-pinecone/internal/gen/control"
+	"github.com/pinecone-io/go-pinecone/internal/util"
 	"io"
 	"net/http"
 )
 
 type Client struct {
 	apiKey     string
 	restClient *control.Client
+	sourceTag  string
+}
+
+type CustomHeader struct {
+	userAgent string
+}
+
+func NewUserAgentProvider(userAgent string) *CustomHeader {
+	return &CustomHeader{userAgent: userAgent}
+}
+
+func (s *CustomHeader) Intercept(ctx context.Context, req *http.Request) error {
+	req.Header.Set("User-Agent", s.userAgent)
+	return nil
```

Comment:
This approach is based on https://medium.com/@kyodo-tech/go-client-code-generation-from-swagger-and-openapi-a0576831836c

Reply:
Good call, I moved it to an internal `provider` package. I also changed it to be generic to setting any header